crates/router/src/core/payment_methods/cards.rs

```diff
@@ -4669,25 +4653,64 @@ pub async fn get_mca_status(
             .change_context(errors::ApiErrorResponse::MerchantConnectorAccountNotFound {
                 id: merchant_id.get_string_repr().to_owned(),
             })?;
-        let mut mca_ids = HashSet::new();
-        let mcas = mcas
+        let mca_ids = mcas
             .into_iter()
             .filter(|mca| {
-                mca.disabled == Some(false) && profile_id.clone() == Some(mca.profile_id.clone())
+                mca.disabled.is_some_and(|disabled| !disabled)
+                    && profile_id
+                        .as_ref()
+                        .is_some_and(|profile_id| *profile_id == mca.profile_id)
             })
-            .collect::<Vec<_>>();
+            .map(|mca| mca.get_id())
+            .collect::<HashSet<_>>();
 
-        for mca in mcas {
-            mca_ids.insert(mca.get_id());
-        }
-        for mca_id in connector_mandate_details.keys() {
-            if mca_ids.contains(mca_id) {
-                return Ok(true);
-            }
+        if connector_mandate_details
+            .keys()
+            .any(|mca_id| mca_ids.contains(mca_id))
+        {
+            return Ok(true);
         }
     }
     Ok(false)
 }
+
+#[cfg(all(feature = "v2", feature = "payment_methods_v2"))]
+#[allow(clippy::too_many_arguments)]
+pub async fn get_mca_status(
+    state: &routes::SessionState,
+    key_store: &domain::MerchantKeyStore,
+    profile_id: Option<id_type::ProfileId>,
+    merchant_id: &id_type::MerchantId,
+    is_connector_agnostic_mit_enabled: bool,
+    connector_mandate_details: Option<&payment_method::PaymentsMandateReference>,
+    network_transaction_id: Option<&String>,
+    merchant_connector_accounts: Vec<domain::MerchantConnectorAccount>,
+) -> bool {
+    if is_connector_agnostic_mit_enabled && network_transaction_id.is_some() {
+        return true;
+    }
+    if let Some(connector_mandate_details) = connector_mandate_details {
+        let mca_ids = merchant_connector_accounts
```

Comment:
create a private `struct MerchantConnectorAccounts(Vec<MerchantConnectorAccount>)`

and write these as impl function, use this impl function above too.